bsmetadata/preprocessing_utils.py

```diff
@@ -238,3 +240,137 @@ def preprocess(self, examples: Dict[str, List]) -> Dict[str, List]:
                     example_metadata.append(en)
             continue
         return examples
+
+
+class GenerationLengthPreprocessor(MetadataPreprocessor):
+    """An exemplary metadata preprocessor for adding generation length information based on text."""
+
+    def __init__(self, mode):
+        # The length can be calculated for the whole text or for each sentence of a text individually.
+        # We can specify a global length of a TEXT or a local length for each SENTENCE of a text.
+        # Therefore, we provide two different modes: text (global) or sentence (local).
+        self.mode = mode  # {text, sentence}
+
+    def preprocess(self, examples: Dict[str, List]) -> Dict[str, List]:
+        """
+        Iterate through all the examples retrieve the length meta information.
+        """
+        for example_text, example_metadata in zip(examples["text"], examples["metadata"]):
+            if self.mode == "text":
+                text_length = self._extract_length_from_text(example_text)
+                example_length = {"key": "length", "type": "global", "value": text_length}
+            elif self.mode == "sentence":
+                example_length = self._extract_length_from_sentences(example_text)
+            else:
+                print("Please select a valid length type [text or sentence].")
+
+            if not example_length:
+                continue
+
+            example_metadata.append(example_length)
+        examples["metadata"] = (
+            [m[0] for m in examples["metadata"]] if self.mode == "sentence" else examples["metadata"]
+        )  # reformatting of nested lists
+
+        return examples
+
+    def _extract_length_from_text(self, text: str) -> Optional[str]:
+        """
+        Identify the length of a text.
+        """
+
+        return str(len(text))  # char-based length
+
+    def _extract_length_from_sentences(self, text: str) -> Optional[str]:
```

Comment:
Reading the content of the method, I guessed that the output format is a `list` (and not a `str`). Are you agree?